src/plugin/utils.ts

```
@@ -0,0 +1,85 @@
+
+export function generatePluginId(manifest: IPluginManifest): string {
+  return manifest.name.toLowerCase().replace(/\s+/g, '_');
+}
+
+export function sortDrawerItems(items: IDrawerExtension[]): IDrawerExtension[] {
+  return items.sort((a, b) => (a.order || 999) - (b.order || 999));
+}
+
+export function filterByPermissions<
+  T extends { permissions?: string[]; isAdmin?: boolean },
+>(items: T[], userPermissions: string[], isAdmin: boolean = false): T[] {
```

Biome (1.9.4)

[error] This type annotation is trivially inferred from its initialization.

Safe fix: Remove the type annotation.


(lint/style/noInferrableTypes)